Fix for issue #240 and issue #237

piotrtomiak opened the pull request:
Instead of repeating the request every time the server is disposed, which can lead to infinite loop, upload job is being notified when server is being disposed gracefully and can finish cleanly. Additionally, if server object changes, file synchronizer index is being cleaned.

angelozerr commented:
Many thanks @piotrtomiak! I will test your fix and tell you if it works with my case.

angelozerr commented:
@piotrtomiak when you hav etime, could you just update https://github.com/angelozerr/tern.java/wiki/New-and-Noteworthy-0.9.0#asynch-tern-completion if it needs more information 

Thanks!

angelozerr commented:
Great job @piotrtomiak it works now like a charm!

piotrtomiak commented:
> @piotrtomiak when you have time, could you just update https://github.com/angelozerr/tern.java/wiki/New-and-Noteworthy-0.9.0#asynch-tern-completion if it needs more information

Updated. Rephrased some of the sentences and added additional information. Not much to add though ;)  Do you usually give a credit for large contributions like that?

jabby commented:
@angelozerr When you said that you will create a basic UI to customize JSHint, did you mean that configuration will be stored in workspace preferences or in project preferences ? 

I think that read .jshintrc file to customize jshint will be a great feature. In fact when I use tern.java, it's for projects which are build with yeoman. In generators there is commonly .jshintrc files which contain all configuration. It will be painful if users, like me, need to create the same configuration in tern.java.

angelozerr commented:
@jabby please post your question at https://github.com/angelozerr/tern.java/issues/242 Thanks!

vrubezhny commented:
@angelozerr, @piotrtomiak,

It looks like these changes are related to https://issues.jboss.org/browse/JBIDE-19184 (Threadlock when JS autocomplete turned on) - see https://issues.jboss.org/secure/attachment/12387697/threaddump_jdk1.7.0_71_1422011702000.txt attached to the issue. 

I'm able to reproduce the deadlock on Tern.java of versions 0.6.0, 0.7.0 and 0.8.0 up to this change. So, we are harmed with this bug in coming JBossTools 4.2.3 that is planned to use Tern.java/AngularJS-eclipse v. 0.7.1.

Is it possible to pick up this fix into 0.7.1 stream? (I know there are more fixes and improvements related to the issue in 0.9.0 stream, but at least this fix solved the deadlock problem, so it's critical for us to merge it into 0.7.1 stream).

angelozerr commented:
@vrubezhny I'm afraid that it's a little hard to merge some pieces of code to fix your problem  https://issues.jboss.org/browse/JBIDE-19184 

0.9.0 provides an async completion to avoid hanging eclipse https://github.com/angelozerr/tern.java/wiki/New-and-Noteworthy-0.9.0#async-tern-completion

I have wriiten too a wiki page about tern performance problem at https://github.com/angelozerr/tern.java/wiki/Tern-Performance

My goal is to improve tern script path (like exclude pattern files).

Sorry @vrubezhny  -(
